### ISSUE TITLE
Prevent Notification from being shown in NotificationReceived

### ISSUE THREAD

**Issue by Raffro** [TobiasBuchholz/Plugin.Firebase#236]
I have a app with instant messaging function. I want the user to not receive the notifications if a user is active in the chatpage.

Is it possible to use the CrossFirebaseCloudMessaging.Current.NotificationReceived event in order to stop the Notification?
I tried accessing the EventHandler and then called return, but the notification is shown anyway.

**Comment by andyzukunft** [comment]
The issue with `data` is only allowed to contain string values. As such `is_silent_in_foreground: "true"` would be the correct format (or: our C# class needs to be a string and not a boolean).

The Google Api does also not matter (I guess) as `is_silent_in_foreground` is a custom field only applicable for Plugin.Firebase.

@Gekidoku 
I have a short question. Do you have a solution to "notify" the user (ringtone, vibration, depending on the smartphone configuration) if a silent notification is received?

**Comment by Raffro** [comment]
Yeah this is not exactly what I have meant..
The notification is definitely sent, but how to cancel it directly from the app? Is silent in background cant be modified in the received event, because the event args are read only.

If there is no such functionality at the moment i need to decide individuakky on the server whether to send a notification to a specific device or not.

**Comment by andyzukunft** [comment]
I think there might be a miscommunication. If you send your message with the `data` content `is_silent_in_foreground: "true"` the notification will not be displayed by the toolkit if the app is in foreground.

Btw: thanks Gekidoku for clarifying.

**Comment by Raffro** [comment]
So, ist it possibke to catch the notification app-side and decide to display it or not, or not?

**Comment by andyzukunft** [comment]
Yes. The event handler `CrossFirebaseCloudMessaging.Current.NotificationReceived` is triggered in both cases (is_silent_in_foreground = true, is_silent_in_foreground = false) and if is_silent_in_foreground = true you need to handle it manually if you need/want to.

With Plugin.Firebase.CloudMessaging v2.0.4 Local Notification have been implemented. I believe this is what you could use to create / handle if a silent notification is incoming in NotificationReceived and you want to show it manually.
